solutions/agents/variables.tf

```diff
@@ -0,0 +1,222 @@
+}
+
+variable "cluster_config_endpoint_type" {
+  description = "Specify which type of endpoint to use for for cluster config access: 'default', 'private', 'vpe', 'link'. 'default' value will use the default endpoint of the cluster."
+  type        = string
+  default     = "private"
+  nullable    = false # use default if null is passed in
+  validation {
+    error_message = "Invalid Endpoint Type! Valid values are 'default', 'private', 'vpe', or 'link'"
+    condition     = contains(["default", "private", "vpe", "link"], var.cluster_config_endpoint_type)
+  }
+}
+
+##############################################################################
+# Log Analysis variables
+##############################################################################
+
+variable "log_analysis_enabled" {
+  type        = bool
+  description = "Deploy IBM Cloud Logging agent if set as true."
+  default     = true
+}
+
+
+variable "log_analysis_agent_tags" {
+  type        = list(string)
+  description = "List of tags to associate to all log records that the agent collects so that you can identify the agent's data quicker in the logging UI. NOTE: Use the 'log_analysis_add_cluster_name' variable to add the cluster name as a tag."
+  default     = []
+  nullable    = false
+}
+
+variable "log_analysis_add_cluster_name" {
+  type        = bool
+  description = "If true, configure the log analysis agent to attach a tag containing the cluster name to all log messages."
+  default     = true
+}
+
+variable "log_analysis_ingestion_key" {
+  type        = string
+  description = "Ingestion key for the IBM Cloud Logging agent to communicate with the instance."
+  sensitive   = true
+  default     = null
+}
+
+variable "log_analysis_secret_name" {
+  type        = string
+  description = "The name of the secret which will store the ingestion key."
+  default     = "logdna-agent"
+  nullable    = false
+}
+
+variable "log_analysis_instance_region" {
+  type        = string
+  description = "The IBM Log Analysis instance region. Used to construct the ingestion endpoint."
+  default     = "us-south"
+}
+
+variable "log_analysis_endpoint_type" {
+  type        = string
+  description = "Specify the IBM Log Analysis instance endpoint type (public or private) to use. Used to construct the ingestion endpoint."
+  default     = "private"
+  validation {
+    error_message = "The specified endpoint_type can be private or public only."
+    condition     = contains(["private", "public"], var.log_analysis_endpoint_type)
+  }
+}
+
+variable "log_analysis_agent_custom_line_inclusion" {
+  description = "Log Analysis agent custom configuration for line inclusion setting LOGDNA_K8S_METADATA_LINE_INCLUSION. See https://github.com/logdna/logdna-agent-v2/blob/master/docs/KUBERNETES.md#configuration-for-kubernetes-metadata-filtering for more info."
+  type        = string
+  default     = "label.app.kubernetes.io/name:sample-app\\, annotation.user:sample-user"
```

Comment:
What are these values? This should have a default value of `null` and consumer can choose if they want to customise

Reply:
done